cli/src/cmd/cast/access_list.rs

```diff
@@ -0,0 +1,87 @@
+// cast estimate subcommands
+use crate::{
+    opts::{EthereumOpts, TransactionOpts},
+    utils::{self},
+};
+use cast::{Cast, TxBuilder};
+use clap::Parser;
+use ethers::types::{BlockId, NameOrAddress};
+use eyre::WrapErr;
+use foundry_config::Config;
+use std::str::FromStr;
+
+#[derive(Debug, Parser)]
+pub struct AccessListArgs {
+    #[clap(
+        help = "The destination of the transaction.", 
+        value_name = "TO",
+        value_parser = NameOrAddress::from_str
+    )]
+    to: Option<NameOrAddress>,
+
+    #[clap(help = "The signature of the function to call.", value_name = "SIG")]
+    sig: Option<String>,
+
+    #[clap(help = "The arguments of the function to call.", value_name = "ARGS")]
+    args: Vec<String>,
+
+    #[clap(
+        long,
+        help = "Data for the transaction.",
+        value_name = "DATA",
+        value_parser = foundry_common::clap_helpers::strip_0x_prefix,
+        conflicts_with_all = &["sig", "args"]
+    )]
+    data: Option<String>,
+
+    #[clap(flatten)]
+    tx: TransactionOpts,
+
+    #[clap(flatten)]
+    eth: EthereumOpts,
+
+    #[clap(
+        long,
+        short,
+        help = "The block height you want to query at.",
+        long_help = "The block height you want to query at. Can also be the tags earliest, finalized, safe, latest, or pending.",
+        value_name = "BLOCK"
+    )]
+    block: Option<BlockId>,
+
+    #[clap(long = "json", short = 'j', help_heading = "Display options")]
+    to_json: bool,
+}
+
+impl AccessListArgs {
+    pub async fn run(self) -> eyre::Result<()> {
```

Comment:
nit: I'd like a separate function that returns the access list (`Cast::new(&provider).access_list`) so it can be tested.
`run` then simply calls this and prints the result

Reply:
I split the function to one that should be testable. Not sure how to go about an actual unit test with a working provider.

Added an integration test.